### DISCUSSION TITLE
Question: how to add an attribute to a table header cell? (<th>)

### DISCUSSION THREAD

**Discussion by Ambient-Impact** [thephpleague/commonmark#710] [Q&A] [answered]
Hello. I've tried all sorts of variants of the attributes syntax, but none of them get identified as attributes by the parser.

I tried inline:

```
| Episode | Episode title | Description{data-table-column-type="long"}| Cupoid challenge theme | Original stream date |
```

I tried the block syntax both before after the pipe, because why not:

```
| Episode | Episode title | Description{: data-table-column-type="long"} | Cupoid challenge theme | Original stream date |
```

```
| Episode | Episode title | Description|{: data-table-column-type="long"} Cupoid challenge theme | Original stream date |
```

I tried the alignment syntax just in case:

```
| Episode | Episode title | Description(data-table-column-type="long") | Cupoid challenge theme | Original stream date |
```

Can attributes on individual table elements only be added programmatically? This is on 1.6, as we can't yet upgrade to 2.0, by the way.

**Comment by colinodell** [comment]
Try adding a space before the opening `{` like this:

```markdown
| Episode | Episode title | Description {data-table-column-type="long"} | Cupoid challenge theme | Original stream date |
```

I think that should work on both 1.6 and 2.0.

**Comment by Ambient-Impact** [comment]
@colinodell Thanks for getting back to me. Tried it on 1.6.6 and it still just gets output as text in the ```<th>```:

![Screenshot 2021-08-22 at 10-16-34 Edit Wiki page Table cell attribute test Omnipedia](https://user-images.githubusercontent.com/2289006/130358420-61e84a28-2692-4048-9893-db8f3b3898a0.png)

![Screenshot 2021-08-22 at 10-13-17 Table cell attribute test Omnipedia](https://user-images.githubusercontent.com/2289006/130358423-744acfb8-8b6c-414c-bc88-230808feb58b.png)

Any other suggestions I could try?

**Comment by colinodell** [comment, marked as answer]
I ran a similar test on 1.6.6 and it seemed to work correctly:

![image](https://user-images.githubusercontent.com/202034/130359891-18a1a632-5f38-4f37-8d9b-0ba38c8e08dd.png)

Do you have any other extensions enabled?

If you're open to debugging this, try setting a breakpoint in `AttributesInlineParser::parse()` to confirm if it's being called and parsing the attribute information.

**Reply by Ambient-Impact** [reply]
You're going to love this: I thought I had the attributes extension enabled, but it turns out I did not; it's now installed and working correctly! 🤦‍♀
